source/tutorials/learning-journey/packaging-existing-software.md

```diff
@@ -0,0 +1,589 @@
+
+### Build Result
+Check your working directory for the result:
+
+```console
+$ ls
+default.nix hello.nix  result
+```
+
+This result is a symbolic link to a Nix store location containing the built binary; you can call `./result/bin/hello` to execute this program:
+
+```console
+$ ./result/bin/hello
+Hello, world!
+```
+
+Congratulations, you have successfully packaged your first program with Nix!
+
+Next, you'll package another piece of software with external-to-`stdenv` dependencies that present new challenges, requiring you to make use of more `mkDerivation` features.
+
+## Something Bigger
+Now you will package a somewhat more complicated program, `icat`, which allows you to render images in your terminal.
+
+To start, modify the `default.nix` from the previous section by adding a new attribute for `icat`:
+
+```nix
+# default.nix
+let
+  pkgs = import <nixpkgs> { };
+in
+{
+  hello = pkgs.callPackage ./hello.nix { };
+  icat = pkgs.callPackage ./icat.nix { };
+}
+```
+
+Now copy `hello.nix` to a new file, `icat.nix`, and update the `name` attribute in that file:
+
+```nix
+# icat.nix
+{ lib
+, stdenv
+}:
+
+stdenv.mkDerivation {
+  name = "icat";
+
+  src = builtins.fetchTarball {
+	...
+  };
+}
+```
+
+Now to download the source code. `icat`'s upstream repository is hosted on [GitHub](https://github.com/atextor/icat), so you should slightly modify the previous [source fetcher](https://nixos.org/manual/nixpkgs/stable/#chap-pkgs-fetchers), this time using `pkgs.fetchFromGitHub` instead of `builtins.fetchTarball`, updating the argument attribute set to the function accordingly:
+
+```nix
+# icat.nix
+{ pkgs
+, lib
+, stdenv
+}:
+
+stdenv.mkDerivation {
+  name = "icat";
+
+  src = pkgs.fetchFromGitHub {
+    ...
+  };
+}
+```
+
+### Fetching Source from GitHub
+While `fetchTarball` required `url` and `sha256` arguments, more are needed for [`fetchFromGitHub`](https://nixos.org/manual/nixpkgs/stable/#fetchfromgithub).
+
+The source is hosted on GitHub at `https://github.com/atextor/icat`, which already gives the first two arguments:
+- `owner`: the name of the account controlling the repository; `owner = "atextor"`
+- `repo`: the name of the repository to fetch; `repo = "icat"`
+
+You can navigate to the project's [Releases page](https://github.com/atextor/icat/releases) to find a suitable `rev`, such as the git commit hash or tag (e.g. `v1.0`) corresponding to the release you want to fetch. In this case, the latest release tag is `v0.5`.
+
+As in the `hello` example, a hash must also be supplied. This time, instead of using `lib.fakeSha256` and letting `nix-build` report the correct one in an error, you can fetch the correct hash in the first place with the `nix-prefetch-url` command. You need the SHA256 hash of the *contents* of the tarball, so you will need to pass the `--unpack` and `--type sha256` arguments too:
```

Comment:
```suggestion
As in the `hello` example, a hash must also be supplied. This time, instead of using `lib.fakeSha256` and letting `nix-build` report the correct one in an error, you can fetch the correct hash in the first place with the `nix-prefetch-url` command. You need the SHA256 hash of the *contents* of the tarball (as opposed to the hash of the tarball file itself), so you will need to pass the `--unpack` and `--type sha256` arguments too:
```

The details here are more intricate (such as the fact that we need this because GitHub occasionally changes the hash of the tarballs, which not all sites do; or how there's no standard directory hash, Nix uses its own NAR format for that; or how there's also `nix-prefetch-github` (and others), but they're not official), but I don't think this needs to be mentioned here.

Reply:
Also, it would be great if you used one line per sentence, [as described in the nix.dev style guidelines](https://nix.dev/contributing/style-guide#one-line-per-sentence), makes reviewing such parts much nicer :smile: